Data/models/Domain.swift

```diff
@@ -25,8 +25,9 @@ public final class Domain: NSManagedObject, CRUD {
     
     // MARK: - Public interface
     
-    public class func getOrCreate(forUrl url: URL) -> Domain {
-        return getOrCreateInternal(url)
+    public class func getOrCreate(forUrl url: URL, persistent: Bool) -> Domain {
+        let context = persistent ? DataController.viewContext : DataController.viewContextInMemory
+        return getOrCreateInternal(url, context: context, save: true)
```

Comment:
What will happen if we add a bookmark in PBM? The Bookmark will be saved to regular data store, but its children Domain will be in in-memory store, won't that crash or something?

Reply:
Nope I tried it out. When you add a bookmark, it is added in regular data store. But when you try to open it there is no such domain in memory so then everything switches to inMemeory context.